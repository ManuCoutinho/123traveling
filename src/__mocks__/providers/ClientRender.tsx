/* eslint-disable react/display-name */

import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import { ChakraProvider } from '@chakra-ui/react'

const createTestQueryClient = () =>
	new QueryClient({
		defaultOptions: {
			queries: {
				retry: false
			}
		}
	})

export function renderWithClient(children: React.ReactNode) {
	const testQueryClient = createTestQueryClient()
	const { rerender, ...result } = render(
		<ChakraProvider>
			<QueryClientProvider client={testQueryClient}>
				{children}
			</QueryClientProvider>
		</ChakraProvider>
	)
	return {
		...result,
		rerender: (rerenderUi: React.ReactNode) =>
			rerender(
				<QueryClientProvider client={testQueryClient}>
					{rerenderUi}
				</QueryClientProvider>
			)
	}
}

export const customWrapper = () => {
	const testQueryClient = createTestQueryClient()

	return ({ children }: { children: React.ReactNode }) => (
		<QueryClientProvider client={testQueryClient}>
			{children}
		</QueryClientProvider>
	)
}
