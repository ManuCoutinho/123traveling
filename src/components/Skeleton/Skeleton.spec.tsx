import { renderWithClient } from '__mocks__/providers/ClientRender'
import { Skeleton } from '.'

describe('<Skeleton/>', () => {
	it('should match to snapshot ', () => {
		const { container } = renderWithClient(<Skeleton />)
		expect(container).toMatchSnapshot()
	})
})
