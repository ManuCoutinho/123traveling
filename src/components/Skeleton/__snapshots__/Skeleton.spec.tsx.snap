// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton/> should match to snapshot  1`] = `
<div>
  <div
    class="chakra-stack css-1byv5eb"
  >
    <div
      class="chakra-skeleton css-olz4s9"
    />
    <div
      class="css-16uc9oj"
    >
      <div
        class="css-wzpjib"
      >
        <div
          class="chakra-skeleton__group css-1ayfwcb"
        >
          <div
            class="chakra-skeleton css-tl1b3o"
          />
          <div
            class="chakra-skeleton css-12r9lpq"
          />
        </div>
      </div>
      <div
        class="chakra-skeleton css-1wnjkes"
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
